clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerBatch.java

```diff
@@ -218,7 +226,7 @@ private ProducerBatch createBatchOffAccumulatorForRecord(Record record, int batc
                 record.key(), record.value(), record.headers()), batchSize);
         ByteBuffer buffer = ByteBuffer.allocate(initialSize);
         MemoryRecordsBuilder builder = MemoryRecords.builder(buffer, magic(), recordsBuilder.compressionType(),
-                TimestampType.CREATE_TIME, 0L, recordsBuilder.isTransactional());
+                TimestampType.CREATE_TIME, 0L);
```

Comment:
Yes, I attempted to fix this previously, but the logic was incorrect. The builder requires that the producerId and the producer epoch are both set if `isTransactional` is set to true. I felt like this was a useful sanity check, so I changed the logic to pass `isTransactional` at the same time that we set the producer id and epoch. Does that make sense?